src/toil/batchSystems/awsBatch.py

```diff
@@ -0,0 +1,519 @@
+
+        # Get rid of the job definition we are using if we can.
+        self.__destroy_job_definition()
+
+    @retry(errors=[BotoServerError])
+    def __try_terminate(self, aws_id: str) -> None:
+        """
+        Try to terminate an AWS Batch job.
+
+        Succeed if it can't be canceled because it has stopped,
+        but fail if it can't be canceled for some other reason.
+        """
+        # Remember that we killed this job so we don't show it as updated
+        # later.
+        self.killed_job_aws_ids.add(aws_id)
+        # Kill the AWS Batch job
+        self.client.terminate_job(jobId=aws_id, reason='Killed by Toil')
+
+    @retry(errors=[BotoServerError])
+    def __wait_until_stopped(self, aws_id: str) -> None:
+        """
+        Wait for a terminated job to actually stop. The AWS Batch API does not
+        guarantee that the status of a job will be SUCCEEDED or FAILED as soon
+        as a terminate call succeeds for it, but Toil requires that a job that
+        has been successfully killed can no longer be observed to be running.
+        """
+
+        while True:
+            # Poll the job
+            response = self.client.describe_jobs(jobs=[aws_id])
+            jobs = response.get('jobs', [])
+            if len(jobs) == 0:
+                # Job no longer exists at all
+                return
+            job = jobs[0]
+            if job.get('status') and job['status'] in ['SUCCEEDED', 'FAILED']:
+                # The job has stopped
+                return
+            # Otherwise the job is still going. Wait for it to stop.
+            logger.info('Waiting for killed job %s to stop', self.aws_id_to_bs_id.get(aws_id, aws_id))
+            time.sleep(2)
+
+    @retry(errors=[BotoServerError])
+    def __get_or_create_job_definition(self) -> str:
+        """
+        Create, if not already created, and return the ARN for the
+        JobDefinition for this workflow run.
+        """
+        if self.job_definition is None:
+            job_def_spec = {
+                'jobDefinitionName': 'toil-' + str(uuid.uuid4()),
+                'type': 'container',
+                'containerProperties': {
+                    'image': self.docker_image,
+                    # Unlike the Kubernetes batch system we always mount the Toil
+                    # workDir onto the host. Hopefully it has its ephemeral disks mounted there.
+                    # TODO: Where do the default batch AMIs mount their ephemeral disks, if anywhere?
+                    'volumes': [{'name': 'workdir', 'host': {'sourcePath': '/var/lib/toil'}}],
+                    'mountPoints': [{'containerPath': self.worker_work_dir, 'sourceVolume': 'workdir'}],
+                    # Requirements will always be overridden but mist be present anyway
+                    'resourceRequirements': [
+                        {'type': 'MEMORY', 'value': str(math.ceil(max(4, to_mib(self.config.defaultMemory))))},
```

Comment:
4mb feels low, even though this reassuringly says this default's day will never come to pass.  Maybe 64mb?

Reply:
This isn't a default; it's a protection against sending the API out-of-spec requests. They specify 4 MiB as the minimum memory request.

If we want to pad the job memory requests for overhead, we need another mechanism for that.

I'll pull this out into a constant to be more clear what it is for.